activerecord/lib/active_record/locking/optimistic.rb

```diff
@@ -63,8 +63,8 @@ def locking_enabled? #:nodoc:
 
         def increment_lock
           lock_col = self.class.locking_column
-          previous_lock_value = send(lock_col).to_i
-          send(lock_col + "=", previous_lock_value + 1)
+          previous_lock_value = send(lock_col)
```

Comment:
yes. https://github.com/bogdanvlviv/rails/blob/4741af97218aba5ca8f655f62bcd3b9ddda7c88d/activerecord/test/cases/locking_test.rb#L247

i explained it in previous comment https://github.com/rails/rails/pull/28833#discussion_r113808471